src/main/utils/upload.ts

```diff
@@ -1,28 +1,48 @@
-import ElectronStore from 'electron-store';
+import Store, { Schema } from 'electron-store';
 import path from 'path';
 import * as fs from 'promise-fs';
-import mime from 'mime';
-import { ENCRYPTION_CHUNK_SIZE } from '../../config';
-import { dialog } from '@electron/remote';
-import { ElectronFile } from '../types';
+import { FILE_STREAM_CHUNK_SIZE } from '../../config';
+import { ElectronFile, StoreType } from '../types';
 
-const store = new ElectronStore();
+const storeSchema: Schema<StoreType> = {
+    done: {
+        type: 'boolean',
+    },
+    filePaths: {
+        type: 'array',
+        items: {
+            type: 'string',
+        },
+    },
+    collectionName: {
+        type: 'string',
+    },
+};
+
+const store = new Store({
+    name: 'upload-status',
+    schema: storeSchema,
+});
+
+// https://stackoverflow.com/a/63111390
+const getAllFilePaths = async (dirPath: string) => {
+    if (!(await fs.stat(dirPath)).isDirectory()) {
+        return [dirPath];
+    }
 
-const getFilesFromDir = (dirPath: string) => {
     let files: string[] = [];
+    const filePaths = await fs.readdir(dirPath);
 
-    // https://stackoverflow.com/a/63111390
-    const getAllFilePaths = (dirPath: string) => {
-        fs.readdirSync(dirPath).forEach((filePath) => {
-            const absolute = path.join(dirPath, filePath);
-            if (fs.statSync(absolute).isDirectory())
-                return getAllFilePaths(absolute);
-            else return files.push(absolute);
-        });
-    };
+    for (const filePath of filePaths) {
+        const absolute = path.join(dirPath, filePath);
+        files = files.concat(await getAllFilePaths(absolute));
+    }
+
+    return files;
+};
 
-    if (fs.statSync(dirPath).isDirectory()) getAllFilePaths(dirPath);
-    else files.push(dirPath);
+export const getFilesFromDir = async (dirPath: string) => {
```

Comment:
why do we have this wrapper? and can just rename `getAllFilePaths` to `getFilesFromDir` na

Reply:
Because `getAllFilePaths` takes only one `dirPath` and we can have multiple such directories and also the `files` variable wouldn't be needed to pass around. I though it would be easier to read, should I remove it?

Reply:
1. don't both of them take only one `dirPath`?
2. I didn't understand why the files variable would be needed to be passed around if we just call `getAllFilePaths`  directly?

Reply:
Sorry my bad, was reading some different commit entirely. Will remove that wrapper and rename it accordingly.